apollo-router/src/plugins/include_subgraph_errors.rs

```diff
@@ -58,11 +76,40 @@ impl Plugin for IncludeSubgraphErrors {
                                 .entry("service")
                                 .or_insert(sub_name_response.clone().into());
                         }
+                        return response;
                     } else {
                         tracing::info!("redacted subgraph({sub_name_response}) errors");
-                        for error in errors.iter_mut() {
+
+                        for error in response.response.body_mut().errors.iter_mut() {
+                            let mut new_extensions = error.extensions.clone();
+                            match subgraph_config.clone() {
```

Comment:
I don't think we need to clone it.
```suggestion
                            match &subgraph_config {
```